cmd/internal/flowmode/cmd_run.go

```diff
@@ -154,6 +168,17 @@ func (fr *flowRun) Run(configFile string) error {
 		return fmt.Errorf("building tracer: %w", err)
 	}
 
+	// Enable the "telemetry.useOtelForInternalMetrics" Collector feature gate.
+	// Currently, Collector components uses OpenCensus metrics by default.
+	// Those metrics cannot be integrated with Agent Flow,
+	// so we need to always use OpenTelemetry metrics.
+	err = enableOtelFeatureGates(
+		"telemetry.useOtelForInternalMetrics",
+	)
+	if err != nil {
+		return err
+	}
```

Comment:
Have you seen https://github.com/grafana/agent/blob/main/component/otelcol/internal/featuregate/install.go? Why do we need this again here?

Reply:
Thanks for letting me know! I had no idea about this. I suppose that package didn't work for me since it relied on the `AGENT_MODE` env var, which the "agent-flow" binary doesn't use. I deleted the package since I think my code would work for both "agent" and "agent-flow" binaries. A bonus about deleting it is that we also don't rely on `init()` as much.

I also pushed a minor change to move my function for setting the feature gate to the util package, so that there is less repetition.